src/replication.c

```diff
@@ -491,6 +492,46 @@ start_rebuild(void *buf, replica_t *replica, uint64_t data_len)
 	return ret;
 }
 
+/*
+ * Compare time..
+ * results: (T1 > T2) ? T1 : T2
+ */
+static int
+compare_time(struct timespec t1, struct timespec t2)
+{
+	if (t1.tv_sec > t2.tv_sec)
+		return 1;
+
+	if ((t1.tv_sec == t2.tv_sec) &&
+	    (t1.tv_nsec > t2.tv_nsec))
+		return 1;
+	return 0;
+}
+
+static int
+check_for_old_ios(spec_t *spec, struct timespec last)
+{
+	struct timespec diff1, diff2 = { 0 }, now, io_time;
+	int ret = 0;
+	int i = 0;
+
+	clock_gettime(CLOCK_MONOTONIC, &now);
+
+	for (i = 0; i < spec->luworkers; i++) {
+		io_time = io_queue_time[i];
+		if (!(io_time.tv_sec == 0 && io_time.tv_nsec == 0)) {
+			time_diff(io_time, now, diff1);
```

Comment:
lets have the compare_time between io_time and last instead of taking the diff

Reply:
ok